sinks/cortex/cortex.go

```diff
@@ -323,15 +339,33 @@ func (s *CortexMetricSink) FlushOtherSamples(context.Context, []ssf.SSFSample) {
 
 // makeWriteRequest converts a list of samples from a flush into a single
 // prometheus remote-write compatible protobuf object
-func makeWriteRequest(metrics []samplers.InterMetric, tags map[string]string) *prompb.WriteRequest {
+func (s *CortexMetricSink) makeWriteRequest(metrics []samplers.InterMetric, tags map[string]string) (*prompb.WriteRequest, map[counterMapKey]float64) {
 	ts := make([]*prompb.TimeSeries, len(metrics))
+	updatedCounters := map[counterMapKey]float64{}
 	for i, metric := range metrics {
+		if metric.Type == samplers.CounterMetric && s.convertCountersToMonotonic {
+			newMetric, counterKey := s.convertToMonotonicCounter(metric)
+			metric = newMetric
+			updatedCounters[counterKey] = metric.Value
+		}
+
 		ts[i] = metricToTimeSeries(metric, tags)
 	}
 
 	return &prompb.WriteRequest{
 		Timeseries: ts,
+	}, updatedCounters
+}
+
+func (s *CortexMetricSink) convertToMonotonicCounter(metric samplers.InterMetric) (samplers.InterMetric, counterMapKey) {
+	sort.Strings(metric.Tags)
+	key := counterMapKey{
+		name: metric.Name,
+		tags: strings.Join(metric.Tags, "|"),
```

Comment:
Just an observation: we use `,` as the delimiter here, and call this `JoinedTags`:
https://github.com/stripe/veneur/blob/master/server.go#L1021-L1022